src/vs/platform/terminal/common/capabilities/commandDetection/promptInputModel.ts

```diff
@@ -297,19 +301,19 @@ export class PromptInputModel extends Disposable implements IPromptInputModel {
 			}
 		}
 
-		const absoluteCursorY = buffer.baseY + buffer.cursorY;
 		let value = commandLine;
 		let ghostTextIndex = -1;
-		let cursorIndex: number;
-		if (absoluteCursorY === commandStartY) {
-			cursorIndex = this._getRelativeCursorIndex(this._commandStartX, buffer, line);
-		} else {
-			cursorIndex = commandLine.trimEnd().length;
+		if (cursorIndex === undefined) {
+			if (absoluteCursorY === commandStartY) {
+				cursorIndex = this._getRelativeCursorIndex(this._commandStartX, buffer, line);
+			} else {
+				cursorIndex = commandLine.trimEnd().length;
+			}
 		}
 
 		// Detect ghost text by looking for italic or dim text in or after the cursor and
 		// non-italic/dim text in the cell closest non-whitespace cell before the cursor
-		if (absoluteCursorY === commandStartY && buffer.cursorX > 1) {
+		if (this._shellType !== GeneralShellType.PowerShell || absoluteCursorY === commandStartY && buffer.cursorX > 1) {
```

Comment:
Inverting `this._shellType !== GeneralShellType.PowerShell` would make it more clear and future proof:

```
this._shellType === PosixShellType.Zsh || this._shellType === PosixShellType.Fish
```

I'm not sure I understand the change though, and maybe we need to change it for pwsh as well. Here's a case that doesn't work currently which might be the fish case you were testing against?

![image](https://github.com/user-attachments/assets/640591af-2c8c-4918-b041-28d51261b276)